libr/core/panels.c

```diff
@@ -2520,15 +2521,37 @@ static int ioCacheOffCb(void *user) {
 	return 0;
 }
 
-static void updateDisassemblyAddr (RCore *core) {
+static void update_disassembly_or_open (RCore *core) {
 	RPanels *panels = core->panels;
 	int i;
+	bool create_new = true;
 	for (i = 0; i < panels->n_panels; i++) {
 		RPanel *p = getPanel (panels, i);
 		if (check_panel_type (p, PANEL_CMD_DISASSEMBLY, strlen (PANEL_CMD_DISASSEMBLY))) {
 			set_panel_addr (core, p, core->offset);
+			create_new = false;
 		}
 	}
+	if (create_new) {
+		RPanel *panel = getPanel (panels, 0);
+		int x0 = panel->view->pos.x;
+		int y0 = panel->view->pos.y;
+		int w0 = panel->view->pos.w;
+		int h0 = panel->view->pos.h;
+		int threshold_w = x0 + panel->view->pos.w;
+		int x1 = x0 + w0 / 2 - 1;
+		int w1 = threshold_w - x1;
+
+		insertPanel (core, 0, PANEL_TITLE_DISASSEMBLY, PANEL_CMD_DISASSEMBLY, false);
+		RPanel *p0 = getPanel (panels, 0);
+		set_geometry (&p0->view->pos, x0, y0, w0 / 2, h0);
+
+		RPanel *p1 = getPanel (panels, 1);
```

Comment:
what if there's only one panel? wouldnt this be NULL?

Reply:
never. a panel is inserted in a few lines ago as you can see, and panels have at least a panel in it always.